olp-cpp-sdk-core/include/olp/core/client/OlpClientSettingsFactory.h

```diff
@@ -104,9 +102,15 @@ class CORE_API OlpClientSettingsFactory final {
    * @note This only makes sense on platforms which actually keep TCP
    * sockets and connections alive for some time and only if you have
    * max_request_count set to something greater then 1 to allow reusing.
+   *
+   * @param[in] settings The `OlpClientSettings` instance.
+   * @param[in] url The url of the host.
+   * @param[in] callback The user callback will be called when request
+   * completed.
    */
-  static void PrewarmConnection(const OlpClientSettings& settings,
-                                const std::string& url);
+  static void PrewarmConnection(
+      const OlpClientSettings& settings, const std::string& url,
+      boost::optional<http::Network::Callback> callback = boost::none);
```

Comment:
Remove optional as std::function can be invalid, i.e. `callback = nullptr`.

```suggestion
      http::Network::Callback callback = nullptr);
```